renku/core/dataset/dataset.py

```diff
@@ -733,17 +733,30 @@ def update_datasets(
     return imported_datasets_view_models, dataset_files_view_models
 
 
-def show_dataset(name):
+def show_dataset(name: str, tag: Optional[str] = None):
     """Show detailed dataset information.
 
     Args:
-        name: Name of dataset to show details for.
+        name(str): Name of dataset to show details for.
+        tag(str, optional): Tags for which to get the metadata (Default value = None).
 
     Returns:
         dict: JSON dictionary of dataset details.
     """
-    dataset = DatasetsProvenance().get_by_name(name, strict=True)
+    datasets_provenance = DatasetsProvenance()
+    dataset = datasets_provenance.get_by_name(name, strict=True)
```

Comment:
This is not used if tag is not None. We should put it inside the if below.

Reply:
it is used both on line 750 in the if and 752 in the get_all_tags method